domain-models-runtime/src/main/java/org/folio/rest/persist/cache/ConnectionCache.java

```diff
@@ -0,0 +1,157 @@
+package org.folio.rest.persist.cache;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ConnectionCache {
+  private static final Logger LOG = LogManager.getLogger(ConnectionCache.class);
+  private static final String LOGGER_LABEL = "CONNECTION MANAGER CACHE STATE";
+
+  private final List<CachedPgConnection> cache;
+  private final Metrics metrics = new Metrics();
+
+  public ConnectionCache() {
+    cache = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  public void remove(CachedPgConnection connection) {
+    synchronized (cache) {
+      cache.remove(connection);
+      metrics.active--;
+      LOG.debug("Removed connection: {} {}",
+          connection.getTenantId(), connection.getSessionId());
+    }
+  }
+
+  public void tryAdd(CachedPgConnection connection) {
+    synchronized (cache) {
+      if (cache.contains(connection)) {
+        LOG.debug("Item already exists in cache: {} {} {}",
+            connection.getTenantId(), connection.getSessionId(), connection.isAvailable());
+        return;
+      }
+      cache.add(connection);
+    }
+  }
+
+  public void tryRemoveOldestAvailableAndClose() {
+    synchronized (cache) {
+      cache.stream()
+          .filter(CachedPgConnection::isAvailable)
+          .min(Comparator.comparingLong(CachedPgConnection::getLastUsedAt))
+          .ifPresent(connection -> {
+            connection.getWrappedConnection().close();
+            cache.remove(connection);
+            metrics.active--;
+            LOG.debug("Removed and closed oldest available connection: {} {}",
+                connection.getTenantId(), connection.getSessionId());
+          });
+    }
+  }
+
+  public Optional<CachedPgConnection> getConnection(String tenantId) {
+    Optional<CachedPgConnection> connectionOptional;
+    synchronized (cache) {
+      connectionOptional =
+          cache.stream().filter(item ->
+              item.getTenantId().equals(tenantId) && item.isAvailable()).findFirst();
+    }
+    return connectionOptional;
+  }
+
+  public void clear() {
+    synchronized (cache) {
+      cache.clear();
+      metrics.clear();
+    }
+  }
+
+  public int size() {
+    synchronized (cache) {
+      return cache.size();
+    }
+  }
+
+  public void log(String event) {
```

Comment:
log methods on a cache object does not seem intuitive. Can you add some comments here?